src/app/modules/submission/encrypt-submission/encrypt-submission.controller.ts

```diff
@@ -106,6 +125,110 @@ export const handleEncryptedSubmission: RequestHandler = async (
+    encryptedContent,
+  )
+  if (encryptedEncodingResult.isErr()) {
+    logger.error({
+      message: 'Error verifying content has encrypted encoding.',
+      meta: logMeta,
+      error: encryptedEncodingResult.error,
+    })
+    const { statusCode, errorMessage } = mapRouteError(
+      encryptedEncodingResult.error,
+    )
+    return res.status(statusCode).json({
+      message: errorMessage,
+    })
+  }
+
+  // Process encrypted submission
+  const processedResponsesResult = await getProcessedResponses(form, responses)
+  if (processedResponsesResult.isErr()) {
+    logger.error({
+      message: 'Error processing encrypted submission.',
+      meta: logMeta,
+      error: processedResponsesResult.error,
+    })
+    const { statusCode, errorMessage } = mapRouteError(
+      processedResponsesResult.error,
+    )
+    return res.status(statusCode).json({
+      message: errorMessage,
+    })
+  }
+  const processedResponses = processedResponsesResult.value
+  // eslint-disable-next-line @typescript-eslint/no-extra-semi
+  ;(req.body as WithParsedResponses<
+    typeof req.body
+  >).parsedResponses = processedResponses
+  // Prevent downstream functions from using responses by deleting it.
+  delete (req.body as SetOptional<EncryptSubmissionBody, 'responses'>).responses
+
+  // Checks if user is SPCP-authenticated before allowing submission
+  const { authType, _id } = form
+  if (authType === AuthType.SP || authType === AuthType.CP) {
+    const useCpCloud = spcpFeature.props?.cpCloudFormId === String(_id)
+    const spcpResult = await SpcpFactory.extractJwt(
+      req.cookies,
+      authType,
+    ).asyncAndThen((jwt) =>
+      SpcpFactory.extractJwtPayload(jwt, authType, useCpCloud),
+    )
+    if (spcpResult.isErr()) {
+      const { statusCode, errorMessage } = mapRouteError(spcpResult.error)
+      logger.error({
+        message: 'Failed to verify JWT with auth client',
+        meta: logMeta,
+        error: spcpResult.error,
+      })
+      return res.status(statusCode).json({
+        message: errorMessage,
+        spcpSubmissionFailure: true,
+      })
+    }
+    res.locals.uinFin = spcpResult.value.userName
+    res.locals.userInfo = spcpResult.value.userInfo
+  }
+
+  // validating that submitted MyInfo field values match the values
+  // originally retrieved from MyInfo.
+  const uinFin = (res as ResWithUinFin<typeof res>).locals.uinFin
+  const requestedAttributes = form.getUniqueMyInfoAttrs()
+  if (authType === AuthType.SP && requestedAttributes.length > 0) {
+    if (!uinFin) {
+      return res.status(StatusCodes.UNAUTHORIZED).send({
+        message: 'Please log in to SingPass and try again.',
+        spcpSubmissionFailure: true,
+      })
+    }
+    const myinfoResult = await MyInfoFactory.fetchMyInfoHashes(
+      uinFin,
+      formId,
+    ).andThen((hashes) =>
+      MyInfoFactory.checkMyInfoHashes(req.body.parsedResponses, hashes),
+    )
```

Comment:
Could you help to see (maybe in a future PR) whether this can be rolled into a single service?